internal/connect/registry_auth.go

```diff
@@ -0,0 +1,174 @@
+package connect
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// If the configuration file is not found for podman which resides
+// in ${XDG_RUNTIME_DIR}/containers/auth.json it will fall back to
+// docker configuration in ${HOME}/.docker.config.json
+// See: https://docs.podman.io/en/stable/markdown/podman-login.1.html
+
+const (
+	DEFAULT_DOCKER_CLIENT_CONFIG = ".docker/config.json"
+	DEFAULT_PODMAN_CONFIG        = "containers/auth.json"
+	DEFAULT_SUSE_REGISTRY        = "https://registry.suse.com"
+)
+
+// we need this to allow mocking the system function in our tests
+var (
+	readFile  = os.ReadFile
+	writeFile = os.WriteFile
+	userHome  = os.UserHomeDir
+	mkDirAll  = os.MkdirAll
+)
+
+// This is already implemented in containers. But for the fun sake we reinvent the wheel!
+// See: https://github.com/containers/image/blob/main/pkg/docker/config/config.go
+// Theoden: You have no dependency here!
+
+type RegistryAuthConfig struct {
+	AuthConfigs map[string]RegistryAuthentication `json:"auths"`
+	CredHelpers map[string]string                 `json:"credHelpers,omitempty"`
+}
+
+type RegistryAuthentication struct {
+	Auth          string `json:"auth,omitempty"`
+	IdentityToken string `json:"identitytoken,omitempty"`
+}
+
+func newRegistryAuthConfig() *RegistryAuthConfig {
+	return &RegistryAuthConfig{
+		AuthConfigs: map[string]RegistryAuthentication{},
+	}
+}
+
+func (cfg *RegistryAuthConfig) LoadFrom(path string) error {
+	data, err := readFile(path)
+	if err != nil {
+		return err
+	}
+	return json.Unmarshal(data, cfg)
+}
+
+func (cfg *RegistryAuthConfig) SaveTo(path string) error {
+	data, err := json.MarshalIndent(cfg, "", "  ")
+	if err != nil {
+		return err
+	}
+	return writeFile(path, data, 0600)
```

Comment:
Both `podman` and `docker` set `0600` as permissions. I do not think there should be an issue then:
docker: https://github.com/docker/cli/blob/master/cli/config/configfile/file_unix.go#L14
podman: https://github.com/containers/image/blob/main/pkg/docker/config/config.go#L618

But I have no strong opinion, should I set it to `0640`?

Reply:
Actually, the file access mode will not help at all here. I tested with a rootless docker installation and seems like root will not have `XDG_RUNTIME_DIR` set. This is a problem, but something that we need to clearly communicate with the customer.

Reply:
Is this resolved after, we ditched all of the `XDG_RUNTIME_DIR` handling? Now its only `root` which will have a configured `auth.json` which is fine I believe. For ALP it is yet to be declared how most containers will run. Since they aim for system applications in containers, I assume most containers will need root access anyway..

Reply:
My comment is that we need to note somewhere that we don't support rootless/podman setup.